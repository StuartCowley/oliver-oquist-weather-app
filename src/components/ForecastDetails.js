import React from "react";
import PropTypes from "prop-types";

import '../styles/ForecastDetails.css';

function ForecastDetails({ forecast }) {
  const {
    temperature, date, humidity, wind,
  } = forecast;

  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-details">
      <div className="forecast-details__date">
        {formattedDate}
      </div>
      <div className="forecast-details__temperature">
        {temperature.max}
        &deg;C
      </div>
      <div className="forecast-details__temperature">
        {temperature.min}
        &deg;C
      </div>
      <div className="forecast-details__humidity">
        {humidity}
        %
      </div>
      <div className="forecast-details__speed">
        {wind.speed}
        mph
      </div>
      <div className="forecast-details__direction">
        {wind.direction}
      </div>
    </div>
  );
}

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number,
    humidity: PropTypes.number,
    icon: PropTypes.number,
    temperature: PropTypes.shape({
      max: PropTypes.number,
      min: PropTypes.number,
    }).isRequired,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
  }).isRequired,
};

export default ForecastDetails;
