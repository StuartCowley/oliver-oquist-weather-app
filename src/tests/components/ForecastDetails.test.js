import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = [
    {
      date: 1111111,
      humidity: 30,
      temperature: {
        max: 11,
        min: 4,
      },
      wind: {
        speed: 13,
        direction: "s",
      },
    },
  ];

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps[0]} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
