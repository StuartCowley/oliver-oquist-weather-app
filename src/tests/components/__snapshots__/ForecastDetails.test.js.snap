// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__temperature"
    >
      11°C
    </div>
    <div
      class="forecast-details__temperature"
    >
      4°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      30%
    </div>
    <div
      class="forecast-details__speed"
    >
      13mph
    </div>
    <div
      class="forecast-details__direction"
    >
      s
    </div>
  </div>
</DocumentFragment>
`;
